File: Solutions in Javascript/KthSmallestElementInA_BST.js
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
let ith;
let ans;
let isFind;
var inorder = function(node, k){
    if(isFind){
        return;
    }
    if(node === null){
        return;
    }
    inorder(node.left, k);
    ith++;
    if(ith === k){
        ans = node.val;
        isFind = true;
        return;
    }
    inorder(node.right, k);
}
var kthSmallest = function(root, k) {
    ith = 0;
    ans = 0;
    isFind = false;
    inorder(root, k);
    return ans;
};

File: Solutions in Java/KthSmallestElementInA_BST.java
/**
 * @author : Anmol Agrawal
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ith;
    int ans;
    boolean isFind;
    void inorder(TreeNode node, int k){
        if(isFind){
            return;
        }
        if(node == null){
            return;
        }
        inorder(node.left, k);
        ith++;
        if(ith == k){
            ans = node.val;
            isFind = true;
            return;
        }
        inorder(node.right, k);
    }
    public int kthSmallest(TreeNode root, int k) {
        ith = 0;
        ans = 0;
        inorder(root, k);
        return ans;
    }
}